src/main/java/beeboo/components/Storage.java

```diff
@@ -0,0 +1,137 @@
+package beeboo.components;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import beeboo.exception.NoFileException;
+import beeboo.task.Deadlines;
+import beeboo.task.Events;
+import beeboo.task.Tasks;
+import beeboo.task.ToDos;
+/**
+ * The Storage class handles the reading and writing of task data to and from a file.
+ * It provides methods for loading tasks from a file and saving tasks to a file.
+ */
+public class Storage {
+
+    /**
+     * The path to the file used for storing task data.
+     */
+    private final String filePath;
+
+    /**
+     * Constructs a Storage instance with the specified file path.
+     *
+     * @param filePath the path to the file where task data is stored
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads task data from the file specified by the file path.
+     * It reads the file line by line, parses the task data, and creates Task objects.
+     *
+     * @return an ArrayList of Tasks objects loaded from the file
+     * @throws NoFileException if the file does not exist
+     */
+    public ArrayList<Tasks> load() throws NoFileException {
+        File file = new File(filePath);
+        ArrayList<Tasks> list = new ArrayList<>();
+        if (!file.exists()) {
+            throw new NoFileException("");
+        } else {
```

Comment:
Perhaps consider removing the else-condition; if the file doesn't exist and an exception is thrown, the code flow stops. This could help to reduce the level of indentations.